tests/helpers.py

```diff
@@ -120,33 +123,57 @@ def create_pod(
     else:
         pod_dict = pod_dict_path if pod_dict_path else constants.CSI_CEPHFS_POD_YAML
         interface = constants.CEPHFS_INTERFACE
-
+    if dc_deployment:
+        pod_dict = pod_dict_path if pod_dict_path else constants.FEDORA_DC_YAML
     pod_data = templating.load_yaml_to_dict(pod_dict)
-    pod_data['metadata']['name'] = create_unique_resource_name(
+    pod_name = create_unique_resource_name(
         f'test-{interface}', 'pod'
     )
+    pod_data['metadata']['name'] = pod_name
     pod_data['metadata']['namespace'] = namespace
+    if dc_deployment:
+        pod_data['metadata']['labels']['app'] = pod_name
+        pod_data['spec']['template']['metadata']['labels']['name'] = pod_name
+
     if pvc_name:
-        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_name
+        if dc_deployment:
+            pod_data['spec']['template']['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_name
+        else:
+            pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_name
 
     if node_name:
         pod_data['spec']['nodeName'] = node_name
     else:
         if 'nodeName' in pod_data.get('spec'):
             del pod_data['spec']['nodeName']
-
-    pod_obj = pod.Pod(**pod_data)
-    pod_name = pod_data.get('metadata').get('name')
-    created_resource = pod_obj.create(do_reload=wait)
-    assert created_resource, (
-        f"Failed to create resource {pod_name}"
-    )
-    if wait:
-        assert wait_for_resource_state(
-            resource=pod_obj, state=desired_status, timeout=120
+    if sa_name and dc_deployment:
+        pod_data['spec']['template']['spec']['serviceAccountName'] = sa_name
+    if dc_deployment:
+        ocs_obj = create_resource(wait=False, **pod_data)
+        logger.info(ocs_obj.name)
+        assert (ocp.OCP(kind='pod', namespace=namespace)).wait_for_resource(
+            condition=constants.STATUS_SUCCEEDED,
+            resource_name=pod_name + '-1-deploy ',
```

Comment:
Just to be sure, the space at the end of the `resource_name` (after `-1-deploy`) is intended?

Reply:
Done